### ISSUE TITLE
[Wave 03] Application use-case + control-orchestration cut-over for #69~#78

### ISSUE THREAD

**Issue by bonggyulim** [solar-ai-dev/mcp-work-agent#100]
## Fixed branch / dependency
Work only on `refactor/canonical-architecture-migration`.

Start only from the pushed #99 SHA.

```text
#81 → #98 → #99 → #100 → #101 → #102 → #103 → #97
```

Do not start #101 until this issue is complete, tested, committed, and pushed.

## Mission
Cut over the affected **Application use-case/control orchestration** to one canonical production authority per capability, using the #98 Domain and #99 Persistence/Port foundations.

This is not a rewrite of working handlers. Existing use-case logic, replay handling, UoW orchestration, pure mapping/validation, and tests are primary reuse candidates.

Disposition preference:

```text
KEEP
→ MOVE / RENAME / MOVE_RENAME
→ SPLIT / MERGE
→ targeted correction
→ REWRITE only when use-case semantics/transaction ownership make preservation unsafe
→ DELETE only after all intended callers are cut over
→ CREATE only when semantic search proves no reusable implementation exists
```

## Authority
Read synchronized `00 → concern owner → 04-A/04 → 05/06/07/08/09/10/11/15 → 16/01,05,08,11,12,13 → 12-A/12 → Ledger → completed #98/#99 → actual code/tests`.

Application owns use-case semantics and short UoW orchestration. Repository owns CAS/persistence only. Concrete Adapter/SQLite/Provider/checkpoint internals are not Application authority.

## Verified current-code baseline — HEAD before #81
Baseline investigated: `2f344e4b93ace4d36a30fd0b6a8d6b12f1036c9c`. Re-verify after #99 before editing.

### A. StartRun is mostly implemented — KEEP the core and correct specific defects
Current reusable path:

- `api/schemas/runs/start_run.py`
- `api/routes/runs.py::start_run()`
- `application/use_cases/run/start_run.py → StartRunHandler`

Already correct/reusable:
- wire accepts `selected_resource_handles`, not Browser-authored raw Resource identity;
- route resolves each handle through the canonical selection-handle resolver before StartRun;
- StartRun preallocates server-owned Run/Message/workflow/thread/handoff identities;
- one UoW already persists Run, USER Message, Conversation update, selected ResourceRefs, initial WorkflowBinding, START handoff, Audit/Trace, Receipt/Result;
- route schedules only after the handler returns from the committed UoW.

**Disposition: `KEEP + targeted correction`; do not replace StartRun.**

Confirmed corrections:
1. `StartRunHandler` currently evaluates `transition_start_run(...)` before `create_workflow_binding(...)`. Synchronized Source requires initial WorkflowBinding materialization before Domain guard evaluation. Reorder within the same short UoW without changing the external wire.
2. `application/use_cases/run/start_run.py` imports `sqlite3` directly. Remove concrete SQLite ownership from Application; map persistence conflict through #99 abstract Repository/UoW error semantics.
3. selected Resource persistence currently maps Registry resource types through a legacy `_RESOURCE_STORAGE_IDENTITIES` vocabulary (`THREAD/MESSAGE/EVENT/...`). Persist the exact canonical Registry `resource_type`; preserve reusable selection-handle validation/materialization logic.
4. current StartRun Audit identity uses `RUN_CREATED`. Current Audit contract requires `RUN_STARTED`. Correct the event identity without rewriting the handler.
5. preserve commit→post-commit scheduler behavior and durable PENDING truth on scheduling failure.

### B. BeginPlanning is a KEEP candidate with two exact corrections
Current:

- `application/use_cases/run/begin_planning.py`
- #98 `domain/run/transitions/begin_planning.py`

The handler already owns receipt/UoW/CAS/plan-supersession/audit logic and should be preserved.

Confirmed defects:
1. current published Plan lookup accepts `{WAITING_APPROVAL, ACTIVE}`. `ACTIVE` is legacy READ-only and must not be current approval-gated Write Plan authority.
2. `_stage_context_adjustment()` directly constructs `MainControlResumeTargetV2(RETRIEVAL_ENTRY)`. Issue/validate this target through the existing canonical `ResumeTargetRegistry` seam instead.

**Disposition: `KEEP + targeted correction`.** Do not create a second Context Adjustment handler or private target table.

### C. ScheduleRunExecution already has strong admission/binding mechanics — preserve them
Current:

- `application/use_cases/run/schedule_run_execution.py`
- `CheckpointEffectiveBindingResolver`

Already correct/reusable:
- persisted handoff/admission CAS;
- expected Run-version binding;
- existing-admission validation;
- non-ACCEPTED release path;
- `CONSUMED_CONTINUATION_RECOVERY` binding from typed active-lineage checkpoint;
- target legality delegates to injected `ResumeTargetRegistry` validator.

Do **not** reintroduce private profile/stage/target lookup tables. That earlier finding is already closed in the investigated HEAD.

Confirmed remaining scheduler gap:
- terminal/preempting early-return occurs before stale persisted admission retirement in some cases;
- fresh NORMAL scheduling does not centrally reject `REAUTH_REQUIRED | RECOVERY_REQUIRED | CANCEL_REQUESTED` the way consumed-continuation recovery does.

Required correction:
- ScheduleRunExecution itself is the independently interleavable Domain-authority fence;
- inspect/retire stale persisted admission before terminal/preemption return;
- new obsolete NORMAL admission = 0;
- WEP submit = 0 on stale/preempted authority;
- use existing #99 typed release/retirement surfaces, not new raw SQL/status mutation.

### D. Redrive has useful logic but incorrect precedence/liveness/result contract
Current `application/use_cases/run/redrive_workflow_handoffs.py` already contains useful:
- Domain-status fence;
- consumed-continuation scheduling path;
- deterministic blocked-binding recovery fingerprint;
- RequireRecovery→supersede flow.

Production `api/composition.py` **already constructs and injects `RequireRecoveryHandler`**. Do not claim production currently omits the dependency. Preserve that wiring.

Remaining concrete defects:
- constructor still permits `require_recovery=None`, creating a no-op semantic path that must not be a production-capable authority;
- BLOCKED rows are loaded first and can consume the batch before unrelated actionable rows;
- historical `list_redriveable` ordering plus first matching CONSUMED row can select stale lineage rather than the exact current active CONSUMED lineage;
- result only exposes inspected/accepted/blocked counts and is insufficient for exact durable-progress/has-more/no-progress startup semantics.

Required correction:
- exact active CONSUMED lineage/current Domain fence first;
- then BLOCKED_BINDING recovery;
- then PENDING/DISPATCHED dispatch;
- fair bounded selection across Runs/lanes;
- mandatory RequireRecovery production dependency;
- typed `progressed_count`/durable-progress + `has_more/actionable` evidence;
- zero durable progress terminates safely without starving later rows.

Preserve the existing deterministic fingerprint and supersession implementation where valid.

### E. Recovery already has canonical handler files; the problem is duplicate/incomplete authority, not absence
Current canonical candidates exist:

- `application/use_cases/recovery/require_recovery.py → RequireRecoveryHandler`
- `application/use_cases/recovery/resolve_recovery.py → ResolveRecoveryHandler`

`RequireRecoveryHandler` already has reusable RecoveryContext/Audit/receipt-building logic. Keep and centralize it.

`ResolveRecoveryHandler` exists but is incomplete: current implementation does not fully consume persisted RecoveryContext/reason matrix/current child facts/NO_PROGRESS/context settlement/required audit+cleanup+handoff semantics.

More importantly, duplicate semantic writers are live:

- `application/use_cases/run/resume_run.py` directly calls `runs.resolve_recovery(... VERIFYING)` for `RECOVERY_RECHECK`;
- the same `ResumeRunHandler` directly performs `runs.require_recovery(...)`, RecoveryContext persistence and Audit in a reauth dispatch-uncertain branch;
- `api/routes/runs.py:/resolve-recovery` uses legacy `ResolveMismatchRecoveryHandler`, not canonical `ResolveRecoveryHandler`;
- `ResumeRunHandler.from_legacy_service_supplier(...)` reaches private coordinator/query/runtime internals.

Required disposition:
- `MERGE` recovery semantics into the canonical RequireRecovery/ResolveRecovery handlers;
- preserve useful context/audit helpers by making them private/shared implementation under the canonical owner only where they do not create a second semantic writer;
- turn ResumeRun into a thin discriminated caller/orchestrator;
- cut `/resolve-recovery` and `/resume(RECOVERY_RECHECK)` to the same canonical `ResolveRecoveryHandler`;
- delete legacy mismatch/direct mutation authority only after callers are cut over.

Do not rewrite all Resume behavior.

### F. RecordReviewResult canonical writer is absent at the investigated HEAD
`application/use_cases/plan/` currently contains only `__init__.py`; `record_review_result.py` is absent.

Before CREATE, semantically search workflow/review/persistence code for an existing equivalent writer. If found, MOVE/MERGE it. Only if no reusable writer exists, CREATE:

- `application/use_cases/plan/record_review_result.py`
- canonical Command/Result/Handler

It must own Plan/current-Run ownership, Plan version, expected review version, `based_on_action_versions`, current child-fact/freshness validation, short-UoW conditional persistence and `REVIEW_RESULT_RECORDED` Audit. Repository owns CAS only.

### G. Conversation/Message queries and other migrated Application paths
Inspect current `GetRunSnapshot`/Conversation history/read handlers for direct `database_path`, `connection_factory`, `sqlite3`, raw SQL/schema ordering ownership. #99 provides canonical query/Repository boundaries; move correct query mechanics behind them rather than replacing behavior.

## Caller cut-over procedure — mandatory
For every affected capability:

1. enumerate all live production callers and concrete imports/exports;
2. identify reusable current handler/service/helper logic;
3. move/merge/correct semantics into canonical owner;
4. rewire API, LangGraph, composition, background drivers and other callers;
5. preserve/move tests;
6. remove old concrete facade/export/private service bridge only after caller count is zero;
7. fail closure if two semantic writers remain.

A canonical file that exists but delegates the semantic decision to a legacy service does not count as cut over.

## Required tests
At minimum:

- StartRun exact ordering: IDs → binding → guard → atomic UoW → COMMIT → scheduler;
- StartRun selection-handle negatives and raw identity rejection remain passing;
- StartRun exact `RUN_STARTED` Audit;
- ResourceRef exact Registry vocabulary;
- scheduler races against REAUTH/RECOVERY/CANCEL/terminal and stale persisted admission retirement;
- WEP submit = 0 on stale/preempted path;
- active CONSUMED H2 selected over historical H1;
- stable blocked row does not starve unrelated Run;
- no-progress/has-more bounded redrive;
- mandatory RequireRecovery path and blocked-binding exactly-once convergence;
- Context Adjustment Registry issuance + Plan ACTIVE rejection + freshness;
- RequireRecovery single-writer rollback/replay;
- ResolveRecovery full reason matrix, stale context, NO_PROGRESS, cleanup/audit/handoff;
- `/resolve-recovery` and `/resume(RECOVERY_RECHECK)` same owner;
- RecordReviewResult stale Plan/review/Action-version rejection;
- Application concrete SQLite/Adapter/provider dependency = 0 for migrated scope;
- caller/import/export negative architecture tests;
- behavior-preservation regression for reused handlers.

## Done condition
At one pushed revision:

- each affected Application capability has one live semantic authority;
- StartRun core is preserved and exact current defects are corrected;
- scheduler central Domain preemption/stale-admission gap = 0;
- redrive active-lineage/fairness/no-progress gap = 0;
- Recovery entry/resolution semantic writers = 1 each;
- legacy ResolveMismatch/direct recovery mutation callers = 0;
- Context Adjustment direct target construction = 0;
- Plan `ACTIVE` Write authority = 0;
- RecordReviewResult freshness writer closed;
- Application concrete SQLite/Adapter/provider dependency = 0 for migrated scope;
- unnecessary canonical-compatible code/test loss = 0;
- unjustified `DELETE + CREATE` churn = 0;
- targeted Application/transaction/integration/architecture tests PASS.

## Completion report
Report authority/caller census, exact old→canonical mappings, dispositions, reused logic, targeted corrections, all post-cut-over deletions and evidence, every CREATE/REWRITE justification, exact test commands/results, and pushed SHA.
